pkgs/by-name/op/opencl-clang/package.nix

```diff
@@ -2,58 +2,74 @@
   lib,
   stdenv,
   applyPatches,
+  callPackage,
+  buildPackages,
+  targetPackages,
   fetchFromGitHub,
   cmake,
   git,
-  llvmPackages_15,
   spirv-llvm-translator,
-  buildWithPatches ? true,
 }:
 
 let
+  llvmPkgs =
+    let
+      llvmPkgs = callPackage ./llvm {
+        buildLlvmTools = buildPackages.opencl-clang.llvmPkgs.tools;
+        targetLlvmLibraries = targetPackages.opencl-clang.llvmPkgs.libraries or llvmPkgs.libraries;
+        targetLlvm = targetPackages.opencl-clang.llvmPkgs.llvm or llvmPkgs.llvm;
+        version = "15.0.7";
+        sha256 = "sha256-wjuZQyXQ/jsmvy6y1aksCcEDXGBjuhpgngF3XQJ/T4s=";
+      };
+    in
+    llvmPkgs;
+
+  patchesOut = stdenv.mkDerivation {
+    pname = "opencl-clang-patches";
+    inherit version src;
+    # Clang patches assume the root is the llvm root dir
+    # but clang root in nixpkgs is the clang sub-directory
```

Comment:
This would be solved by doing a simpler single‐derivation LLVM build rather than copying the complicated split‐packages stuff.